src/ObjectGraph.es6.js

```diff
@@ -181,9 +182,34 @@ ObjectGraph.prototype.rewriteName = function(name) {
 
 // Visit the prototype of o, given its dataMap.
 ObjectGraph.prototype.visitPrototype = function(o, dataMap) {
-  this.storeProto(uid.getId(o), this.visitObject(o.__proto__));
+  this.storeProto(uid.getId(o), this.visitObject(o.__proto__, {proto: true}));
```

Comment:
`visitObject` shouldn't know or care whether or not the entity it operates over is a prototype of another entity.